Cargo.toml

```
@@ -29,6 +29,12 @@ features = [
 ]
 
 [dev-dependencies]
+# TODO: Investigate why using `futures-channel` does not work.
+# In theory, there is no reason why `futures` is required here.
+# The `futures-channel` is sufficient for the test cases, but
+# the code does not compile unless we use the channels from the
+# `futures` crate, which are just re-exported from `futures-channel`!
+futures = { version = "0.3.21", default-features = false, features = ["std"] }
```

For some _strange_ reason, using `futures-channel` does not work here. The error mentions that `UnboundedSender` does not implement the `Sink` trait, which is simply impossible because the `futures-channel` [explicitly implements it](https://docs.rs/futures-channel/latest/futures_channel/mpsc/struct.UnboundedSender.html#impl-Sink%3CT%3E)!

The workaround that I've found is to use the `mpsc::unbounded` channels through the full `futures` _only_ when we run the tests. What I can't figure out is why this workaround even works! In the `futures` crate, the `futures::channel` module literally just re-exports from `futures-channel`. Therefore, there should be no reason for us to use `futures` crate at all because `futures-channel` is already sufficient for running the tests.

Anyway, the `futures` workaround ended up working. We still reap the benefits of shortened compilation times in CI because I explicitly disabled the compilation of `futures-executor` (by adding `default-features = false`). This workaround is irrelevant to the user because Cargo ignores `dev-dependencies` when consuming crates.

---

Very strange indeed. 🤔

I think I can get rid of the `futures-channel` dependency altogether though. I can make a simpler "channel" implementation implements `Stream` and `Sink` using a `VecDeque`, similar to what I did for `AsyncRead`/`AsyncWrite` with [`ByteChannel`](https://github.com/MattiasBuelens/wasm-streams/blob/v0.2.2/tests/util/byte_channel.rs). Then I can use that in the tests instead.

I'll land it as it is now, and remove the dependency in another PR. 😉

---

See #12 for the follow-up.

---

After a good night's sleep, I finally figured out what went wrong here. The `futures` crate enables the `futures-channel/sink` feature by default, which in turn enables the `Sink` trait implementation. But, since this is not a default feature for the bare `futures-channel`, then `UnboundedSender` indeed does not implement the `Sink` trait _yet_.

I figured that I can't possibly leave this unresolved. For anyone interested in reading this PR comment in the future, you're welcome! 🎉
```suggestion
futures-channel = { version = "0.3.21", features = ["sink"] }
```